File: deliveryRoute/src/main/java/com/sparta/deliveryroute/application/dto/RouteResult.java
package com.sparta.deliveryroute.application.dto;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;

import java.util.ArrayList;
import java.util.List;

public record RouteResult(
		@JsonProperty("distance")
		Integer distance,
		@JsonProperty("time")
		Integer time,
		@JsonProperty("path")
		@JsonDeserialize(as = ArrayList.class, contentAs = String.class)
		List<String> path
) {
}

File: hub/src/main/kotlin/com/sparta/hub/application/dto/RouteResult.kt
package com.sparta.hub.application.dto

import com.fasterxml.jackson.annotation.JsonProperty
import com.sparta.hub.presentation.api.response.HubRouteResponseDto

data class RouteResult(
    @JsonProperty("distance")
    val distance: Int?,
    @JsonProperty("time")
    val time: Int?,
    @JsonProperty("path")
    val path: List<String>,
)

fun RouteResult.toDto() = HubRouteResponseDto(
    startHubName = path.first(),
    endHubName = path.last(),
    estimatedSecond = time!!,
    estimatedMeter = distance!!,
)